torch/fx/passes/infra/partitioner.py

```diff
@@ -55,7 +56,44 @@ def __is_node_supported(self, node: Node) -> bool:
             self.operator_support.is_node_supported(dict(self.graph_module.named_modules()), node)
         )
 
+    # This function iterates over all the nodes in the graph and starts a new DFS traversal
+    # for each node. During these traversals it maintains two dictionaries one which keeps
+    # track of all the nodes that are encountered in a DFS traversal starting at a particular
+    # node and another dictionary which keeps track of all the nodes that can reach a particular node.
+    def _cache_dfs_path(self):
+        dfs_paths_from_node = defaultdict(set)
+        dfs_paths_to_node = defaultdict(set)
+        stack = deque()
+
+        for node in self.graph_module.graph.nodes:
```

Comment:
I think there are better way to build the dependency map. 

You should be able to build the map by just traversing the graph in topological order twice, once in forward order, once in reversal.

Reply:
upstream[node] = union(node's upstreams' upstream)
downstream[node] = union(node's downstreams' downstream)

Reply:
Updated. This was a great idea.